core/executor/src/allocator.rs

```diff
@@ -200,45 +194,57 @@ impl FreeingBumpHeapAllocator {
 		Ok(res)
 	}
 
-	fn le_bytes_to_u32(arr: [u8; 4]) -> u32 {
-		u32::from_le_bytes(arr)
-	}
-
-	fn write_u32_into_le_bytes(bytes: u32, slice: &mut [u8]) {
-		slice[..4].copy_from_slice(&bytes.to_le_bytes());
-	}
-
 	fn get_item_size_from_index(index: usize) -> usize {
 		// we shift 1 by three places, since the first possible item size is 8
 		1 << 3 << index
 	}
 
-	fn get_heap_4bytes(&mut self, ptr: u32) -> Result<[u8; 4]> {
-		let mut arr = [0u8; 4];
-		self.heap.get_into(self.ptr_offset + ptr, &mut arr)?;
-		Ok(arr)
+	// Read a u64 from the heap. Used to read heap allocation prefixes.
+	//
+	// u64 is encoded in memory in little-endian form with all bits flipped. Since the heap
+	// allocation prefixes are integers in a low range (eg. they fit in a u8 or u32), the bit flip
+	// serves to make the high order bits 1 instead of 0. This acts as a simple error detection
+	// mechanism, as random bits in the heap space are assumed more likely to be 0 than 1.
```

Comment:
Hmm, I don't see any advantage of all this bit flipping. Where would we use this error detection and how? And to assume that random bits are rather 0 than 1, I don't think that this is correct either. If you want to make the values really unique, we could use something like `0xDEADBEEF` or any other recognizable hex value.

Reply:
FWIW, see the prior discussion [here](https://github.com/paritytech/substrate/pull/3610#discussion_r324264261).

Reply:
Removed.